test/core/libp2p/transport/tcp_edge_cases.cpp

```diff
@@ -55,34 +58,75 @@ TEST(TCP, SingleListenerCanAcceptManyClients) {
   const int kClients = 4;
   const int kSize = 1500;
   const int kRetries = 10;
-
   size_t counter = 0;  // number of answers
+
+  struct Echo : public std::enable_shared_from_this<Echo> {
```

Comment:
Explain the purpose of this struct in docs

Reply:
Ok